lib/newspaper_works/ingest.rb

```diff
@@ -8,5 +11,22 @@
 module NewspaperWorks
   # Module for Ingest adapters that import files into model objects
   module Ingest
+    # Get Geonames URI for closest place match
+    #   Requires Qa::Authorities::Geonames.username is set, likely via
+    #   `Hyrax.config.geonames_username=` setter in
+    #   config/initializers/hyrax.rb of consuming app.
+    # @param place_name [String] Name of place as human-readable text
+    # @return [String, NilClass] URI to Geonames RDF or nil
+    def self.geonames_place_uri(place_name)
+      username = Qa::Authorities::Geonames.username
+      return if username.nil? || username.empty?
+      query = URI.encode(place_name)
+      geo_qs = "q=#{query}&username=#{username}"
+      url = "http://api.geonames.org/search?#{geo_qs}"
+      resp = Faraday.get url
+      doc = Nokogiri.XML(resp.body)
+      geonames_id = doc.xpath('//geonames/geoname[1]/geonameId').first.text
+      "http://sws.geonames.org/#{geonames_id}/about.rdf"
```

Comment:
It might be worth adding some additional error handling here, if the `resp.body` isn't what's expected. If `doc.xpath('//geonames/geoname[1]/geonameId')` returns `nil`, then calling `.first.text` on that will throw a `NoMethodError`?

Also, the canonical Geonames URI is just `http://sws.geonames.org/#{geonames_id}/`, so I think we can leave out the `about.rdf` at the end. That's also how Hyrax stores it, see https://nurax-dev.curationexperts.com/concern/generic_works/9w032309z.json for an example.

Reply:
I have changed this in 5f54a0aa49286b17304f8ec9424b06124151c21a